File: designPatter/Strategy/C++/MallardDuck.cpp
#include "MallardDuck.h"
#include "FlyWithWings.h"
#include "Quack.h"
#include <iostream>

using namespace std;

Mallard::Mallard()
{
	this->setFlyBehavior(new FlyWithWings());
	this->setQuackBehavior(new Quack());
}

void Mallard::display()
{
	cout << "Eu sou o Pato Mallard!!! ;)" << endl;
}

File: designPatter/Strategy/examples/ducks/PHP/Duck/MallardDuck.php
<?php
include_once 'Duck.php';
include_once 'FlyWithWings.php';
include_once 'Quack.php';

class MallardDuck extends Duck
{
	function __construct()
	{
		$this->setQuackBehavior(new Quack());
		$this->setFlyBehavior(new FlyWithWings());
	}

	function display()
	{
		echo "Eu sou o pato MALLARD!!! <br>";
	}

}

?>
